### PR TITLE
Fix SCPEnvelope handling and re-enable quorums

### LINKED ISSUE #518: Quorum consensus isn't being properly reached
https://github.com/bpfkorea/agora/blob/a99d96a5dff959fd4c32b9dc24a837d47bd76713/source/agora/test/Base.d#L642

This was setting the `threshold` to `1` because that's its default value, which means that a node only cared about its own opinion while reaching consensus. Adding additional logging helped confirm this.

Setting it to a proper value however causes consensus failure. It seems that when node A tries to communicate with the other nodes through SCP, the other nodes B, C never receive any messages at all. So it might be an issue with the networking.

I'm investigating.

### PR DESCRIPTION
**Pull request by AndrejMitrovic**
This seems to work for me locally.

### CODE REVIEW

**Comment by AndrejMitrovic** [comment]
Github CI is already green. That's a good sign.

**Comment by AndrejMitrovic** [comment]
@Geod24 it's green. Can you test this out with your PR that was failing?

File: source/agora/network/NetworkClient.d [hunk 1]
```diff
@@ -161,25 +161,26 @@ class NetworkClient
...
+            {
+                this.attemptRequest(this.api.receiveEnvelope(envelope),
+                    null);
+            });
         }
         catch (Exception ex)
         {
-            log.trace("Failed to send envelope: {}", ex.msg);
-            return false;
+            assert(0, "attemptRequest should have cought it");
```
Comment (AndrejMitrovic) [inline source/agora/network/NetworkClient.d]:
```suggestion
            assert(0, "attemptRequest should have caught it");
```

File: source/agora/network/NetworkClient.d [hunk 2]
```diff
@@ -161,25 +161,26 @@ class NetworkClient
 
     /***************************************************************************
 
-        Sends an envelope, and returns a status code if the target node
-        has processed the envelope with no errors (todo: this is blocking..)
```
Comment (AndrejMitrovic) [inline source/agora/network/NetworkClient.d:165]: I like how there was a `todo` here. 🤦‍♂

**Review by Geod24 — APPROVED** [review]
It can't hurt, can you just fix the typo ?

**Comment by AndrejMitrovic** [comment]
Rebased, waiting for green.

**Pull request merged**